### PR TITLE
Fix intermittent validator_exit test failure

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem
Due to shared resources between `validator_exit` and `validator_parallel_exit`, when they are executed in parallel by the rust test executor, these tests hangs. 

#### Summary of Changes
Use one test function to run both tests.

Fixes #

### CODE REVIEW

**Comment by t-nelson** [comment]
Can you remove (or move to a separate commit) the code shuffle in https://github.com/solana-labs/solana/pull/23594/commits/9f56e8905ede504550db54564c560c76a5d18ba8?  It hides the logical changes

File: core/src/validator.rs [hunk 1]
```diff
@@ -1927,6 +1925,12 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_validator_exit() {
+        validator_exit();
+        validator_parallel_exit();
```
Comment (t-nelson) [inline core/src/validator.rs:1931]: these are pretty quick on their own, right? no concern with serialization contributing too negatively on total run time?

**Review by t-nelson — APPROVED** [review]
lgtm. just a curiosity to entertain

File: core/src/validator.rs (hunk 1 above)
Comment (HaoranYi) [reply core/src/validator.rs:1931]: Yeah. Those tests are pretty quick. Around 100ms.

**Comment by jstarry** [comment]
Is it true that these two tests actually share resources? From my understanding they shouldn't be. I think we should revert this change in favor of the real fix (https://github.com/solana-labs/solana/pull/24007 + https://github.com/solana-labs/solana/pull/24036)

**Comment by HaoranYi** [comment]
@jstarry Sure. Good point. I will create another pull request to revert this change. Thanks!

**Pull request merged**